torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -609,9 +666,14 @@ def _should_free_in_backward(
     Returns whether FSDP should free the unsharded flattened parameter in the
     post-backward or not.
     """
-    return (
-        state._sync_gradients and handle.uses_sharded_strategy
-    ) or handle._config.sharding_strategy == HandleShardingStrategy.FULL_SHARD
+    # We always free if we are syncing gradients (i.e. not in no_sync) and parameters
+    # are sharded.
+    free_unsharded = state._sync_gradients and handle.uses_sharded_strategy
+    # For NO_SHARD we don't need to free full parameters, for Zero-2 strategies, we skip
+    # freeing in backward. Note that this also applies for _HYBRID_SHARD_ZERO2.
```

Comment:
nit: This note seems pretty clear? Otherwise, which are the ZeRO-2 _strategies_ (plural)?
```suggestion
    # freeing in backward. Note that this also applies for _HYBRID_SHARD_ZERO2.
```

Reply:
Shard grad op and _HYBRID_SHARD_ZERO2 right?

Reply:
What I mean is that you already said in the comment "for Zero-2 strategies" (plural), and (at least to me) `SHARD_GRAD_OP` is the obvious first one. Then, since you said plural "strategies", there must be a second one, and the second one is probably `_HYBRID_SHARD_ZERO2` since that has `ZERO2` in the name.

This is just a nitpick saying in my opinion we do not need to clarify with the "Note that this also applies for _HYBRID_SHARD_ZERO2" since the preceding sentence is already clear.